flake.nix

```diff
@@ -805,6 +805,24 @@
               '';
             };
 
+            mdbook-check = stdenv.mkDerivation {
+              name = "mdbook-check";
+              dontUnpack = true;
+              buildInputs = [ all-directories-and-files mdbook ];
+              installPhase = ''
+                mkdir -p $out/book
+                chmod 777 $out/book
+                mdbook --version
+                cd ${all-directories-and-files}/book
+                TMPDIR=$out/book mdbook build --dest-dir=$out/book 2>&1 | tee $out/log
```

Comment:
please leave comment why we do not use check

Reply:
done.